pkg/errors/errors_test.go

```diff
@@ -0,0 +1,33 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import (
+	"testing"
+
+	"gotest.tools/assert"
+)
+
+func TestNewInvalidCRD(t *testing.T) {
+	err := newInvalidCRD("serving.knative.dev")
+	assert.Error(t, err, "no Knative serving API found on the backend. Please verify the installation.")
+
+	err = newInvalidCRD("serving")
```

Comment:
But “serving” would be a valid API group name, no? Why not use something like “fake-seving”?

Reply:
this function just creates an error struct based on the api group string that is sent as argument - those tests verifies if the api name was properly parsed.